addons/mail/static/tests/new/activity_menu/activity_menu_tests.js

```diff
@@ -0,0 +1,27 @@
+/** @odoo-module **/
+
+import { click, start, startServer } from "@mail/../tests/helpers/test_utils";
+import { getFixture } from "@web/../tests/helpers/utils";
+
+const ACTIVITY_MENU_SELECTOR = ".o_menu_systray .dropdown-toggle:has(i[aria-label='Activities'])";
+const ACTIVITY_MENU_COUNTER_SELECTOR = `${ACTIVITY_MENU_SELECTOR} .badge`;
+let target;
+
+QUnit.module("activity menu", {
+    async beforeEach() {
+        target = getFixture();
+    },
+});
+
+QUnit.test("should update activities when opening the activity menu", async (assert) => {
+    const pyEnv = await startServer();
+    await start();
+    assert.strictEqual($(target).find(ACTIVITY_MENU_COUNTER_SELECTOR).text(), "");
+    const resPartnerId1 = pyEnv["res.partner"].create({});
+    pyEnv["mail.activity"].create({
+        res_id: resPartnerId1,
+        res_model: "res.partner",
+    });
+    await click(ACTIVITY_MENU_SELECTOR);
+    assert.strictEqual($(target).find(ACTIVITY_MENU_COUNTER_SELECTOR).text(), "1");
```

Comment:
`const` are worthless: `o_menu_systray i[aria-label='Activities']` is good enough.
You can introduce a classname if you feel like it. It's not great to read test when using variables everywhere. Better keep things as plain as possible here.